ssl/ssl_lib.cc

```diff
@@ -3315,3 +3316,34 @@ int SSL_CTX_set1_curves_list(SSL_CTX *ctx, const char *curves) {
 int SSL_set1_curves_list(SSL *ssl, const char *curves) {
   return SSL_set1_groups_list(ssl, curves);
 }
+
+size_t SSL_client_hello_get0_ciphers(SSL *ssl, const unsigned char **out) {
+  STACK_OF(SSL_CIPHER) *client_cipher_suites = SSL_get_client_ciphers(ssl);
+  if (client_cipher_suites == nullptr) {
+      return 0;
+  }
+
+  size_t num_ciphers = sk_SSL_CIPHER_num(client_cipher_suites);
+  if (out != nullptr) {
+    // Called before
+    if(!ssl->client_cipher_suites_arr.empty()) {
+      *out = reinterpret_cast<unsigned char*>(ssl->client_cipher_suites_arr.data());
```

Comment:
NP: I think this would be an interesting sanity test to check in our tests but not necessarily fail in production.
```suggestion
      assert(ssl->client_cipher_suites_arr.size() == num_ciphers);
      *out = reinterpret_cast<unsigned char*>(ssl->client_cipher_suites_arr.data());
```